### PR TITLE
Execute integ_testcases enabling pipelined_managed_tbl0 #6947

### PR DESCRIPTION
**Pull request by prabinakpattnaik**
Signed-off-by: prabina pattnaik <prabinak@wavelabs.ai>

## Summary

Integrated the #5217 and #5685.

## Test Plan

Executed the s1aptest cases with enabled the flag 
pipelined_managed_tbl0: true

Table 0 entries added using RYU controller.
sudo ovs-ofctl -O OpenFlow13 dump-flows gtp_br0 table=0
 cookie=0x0, duration=8.038s, table=0, n_packets=0, n_bytes=0, priority=65525,tun_id=0x64,in_port="g_8d3ca8c0" actions=set_field:02:00:00:00:00:01->eth_src,set_field:ff:ff:ff:ff:ff:ff->eth_dst,set_field:0x64->reg9,set_field:0x9535ed->metadata,resubmit(,1)
 cookie=0x0, duration=8.035s, table=0, n_packets=0, n_bytes=0, priority=65525,ip,in_port=LOCAL,nw_dst=192.168.128.11 actions=set_field:0xc8->tun_id,set_field:192.168.60.141->tun_dst,set_field:0x3->reg8,set_field:0x64->reg9,set_field:0x9535ed->metadata,resubmit(,1)
 cookie=0x0, duration=8.035s, table=0, n_packets=0, n_bytes=0, priority=65525,ip,in_port=mtr0,nw_dst=192.168.128.11 actions=set_field:0xc8->tun_id,set_field:192.168.60.141->tun_dst,set_field:0x3->reg8,set_field:0x64->reg9,set_field:0x9535ed->metadata,resubmit(,1)
 cookie=0x0, duration=8.035s, table=0, n_packets=0, n_bytes=0, priority=65525,arp,in_port=LOCAL,arp_tpa=192.168.128.11 actions=set_field:0x9535ed->metadata,resubmit(,1)
 cookie=0x0, duration=8.035s, table=0, n_packets=0, n_bytes=0, priority=65525,arp,in_port=mtr0,arp_tpa=192.168.128.11 actions=set_field:0x9535ed->metadata,resubmit(,1)
 cookie=0x0, duration=466.079s, table=0, n_packets=1, n_bytes=70, priority=0,in_port=ipfix0 actions=resubmit(,201)
 cookie=0x0, duration=466.079s, table=0, n_packets=0, n_bytes=0, priority=0,in_port="li_port" actions=resubmit(,202)

### CODE REVIEW

**Comment by pshelar** [comment]
This PR needs rebase due to conflict on latest master.

**Comment by pshelar** [comment]
Can you post config changes done to run the integ tests with pipelined tbl0 controller.

**Comment by prabinakpattnaik** [comment]
> This PR needs rebase due to conflict on latest master.

Rebase done with latest master.

**Comment by prabinakpattnaik** [comment]
> Can you post config changes done to run the integ tests with pipelined tbl0 controller.

Following config parameters need to change:
configs/spgw.yml:
pipelined_managed_tbl0: true

configs/pipelined.yml:
5G_feature_set:
 enable: True
ovs_multi_tunnel: True    [for 4G call]

**Review by pshelar — APPROVED** [review]
LGTM

File: lte/gateway/python/magma/pipelined/rpc_servicer.py [hunk 1]
```diff
@@ -472,9 +474,100 @@ def UpdateUEState(self, request, context):
             return UESessionContextResponse(operation_type=request.ue_session_state.ue_config_state,
                                             cause_info=CauseIE(cause_ie=CauseIE.REQUEST_REJECTED_NO_REASON))
 
+
     def _setup_pg_tunnel_update(self, request: UESessionSet, fut: 'Future(UESessionContextResponse)'):
-        res = self._classifier_app.process_mme_tunnel_request(request)
-        fut.set_result(res)
+        cause_ie = CauseIE.REQUEST_ACCEPTED
```
Comment (koolzz) [inline lte/gateway/python/magma/pipelined/rpc_servicer.py]: I actually really preffer the
```
res = self._classifier_app.process_mme_tunnel_request(request)
fut.set_result(res)
```
approach.

The rpc_servier.py should only check if the service is enabled/log grpc payload(`self._log_grpc_payload(request)`). The actual logic is controller speicifc and as such should reside in classifier.py

Should we also add a check to verify the service is running? 
```
        if not self._service_manager.is_app_enabled(
                CLASSIFIER.APP_NAME):
            context.set_code(grpc.StatusCode.UNAVAILABLE)
            context.set_details('Service not enabled!')
            return None
```

**Review by koolzz — CHANGES_REQUESTED** [review]
All logic looks good but please pull the logic back into classifier.py and I'll approve

File: lte/gateway/python/magma/pipelined/rpc_servicer.py (hunk 1 above)
Comment (prabinakpattnaik) [reply lte/gateway/python/magma/pipelined/rpc_servicer.py]: Changes done as requested.

Comment (prabinakpattnaik) [reply lte/gateway/python/magma/pipelined/rpc_servicer.py]: def UpdateUEState(self, request, context):

        self._log_grpc_payload(request)
        if not self._service_manager.is_app_enabled(
              Classifier.APP_NAME):
            context.set_code(grpc.StatusCode.UNAVAILABLE)
            context.set_details('Service not enabled!')
            return None
This check is present in GRPC service "UpdateUEState.

**Review by koolzz — APPROVED** [review]

**Pull request merged**